sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -175,6 +175,30 @@ namespace Azure { namespace Core { namespace Http {
     NetworkAuthenticationRequired = 511, ///< HTTP 511 Network Authentication Required.
   };
 
+  /**
+   * @brief Defines a range of bytes within an HTTP resource, starting at an `Offset` and ending at
+   * `Offset + Length - 1` inclusively.
+   *
+   */
+  struct Range
+  {
+    /**
+     * @brief The starting point of the HTTP Range.
+     *
+     */
+    int64_t Offset = 0;
+
+    /**
+     * @brief The size of the HTTP Range.
+     *
+     */
+    int64_t Length = 0;
+  };
+
+  namespace Internal {
+    std::string RangeToString(Range range);
```

Comment:
It makes the most sense to have ToString on the HttpRange class.  I don't see a good reason to not provide that method.

Reply:
It would be each Service the responsible for adding a toString implementation. Since the way the want to add the header could change.
If it always return "bytes=x-y" and in the future another service want a different unit, or things like that, we can add extra constructors, but toString() would be hard to update

Reply:
Also, if we make it public, why would an end-user want to call toString() anyway?
This was supposed to be Internal only.